src/mongo/platform/strnlen.h

```diff
@@ -0,0 +1,23 @@
+/*    Copyright 2014 MongoDB Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#ifdef MONGO_HAVE_STRNLEN
+#include <string.h>
+#else
+#include <stddef.h>
+extern "C" size_t strnlen(const char *s, size_t maxlen);
```

Comment:
Rather than making this extern and doing this globally, should we just import the name into the mongo namespace:

#include <cstring>
#include <cstddef>

namespace mongo {
#ifdef MONGO_HAVE_STRNLEN
    using ::strnlen;
#else
    size_t strnlen(const char* s, size_t maxlen);
#endif
} // namespace mongo